plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java

```diff
@@ -618,15 +684,15 @@ public List<PartitionColumn> getColumns()
     private static class PartitionColumn
     {
         private final PartitionField field;
-        private final int sourceChannel;
+        private final List<Integer> sourceChannels;
```

Comment:
I find the name a bit misleading - this list is actually a path to the channel - the partition column does not have multiple source channels.